api/core/v1alpha1/authentication_defaults.go

```diff
@@ -79,7 +79,7 @@ func ValidateIdp(idp IdentityProvider, fldPath *field.Path) field.ErrorList {
 // isLowerCaseLetter checks if the given string is a lowercase letter.
 func isLowerCaseLetter(s string) bool {
 	for _, r := range s {
-		if !(unicode.IsLetter(r) && unicode.IsLower(r)) {
+		if !unicode.IsLetter(r) && !unicode.IsLower(r) {
```

Comment:
`!(p && q)` becomes `!p || !q`. But shouldn't we keep this as it is, with a nolint ?